src/FffGcodeWriter.cpp

```diff
@@ -868,9 +876,17 @@ void FffGcodeWriter::addMeshPartToGCode(SliceDataStorage& storage, SliceMeshStor
         mesh->getSettingAsFillMethod("top_bottom_pattern_0") :
         mesh->getSettingAsFillMethod("top_bottom_pattern");
     int skin_angle = 45;
-    if ((skin_pattern == EFillMethod::LINES || skin_pattern == EFillMethod::ZIG_ZAG) && layer_nr & 1)
+    if (skin_pattern == EFillMethod::LINES || skin_pattern == EFillMethod::ZIG_ZAG)
     {
-        skin_angle += 90; // should coincide with infill_angle (if both skin and infill are lines) so that the first top layer is orthogonal to the last infill layer
+        if (mesh->infill_angles.size() > 0)
+        {
+            // skin line angles will follow the same sequence as the infill line angles
+            // this should coincide with infill_angle so that the first top layer is orthogonal to the last infill layer
+            // but if combined_infill_layers above is > 1 then the skin angles and the infill angles will be different
+            skin_angle = mesh->infill_angles.at(layer_nr % mesh->infill_angles.size());
+        }
+        else if(layer_nr & 1)
```

Comment:
Code style: please add a space after `if`.

Please always add braces, which should be on new lines. 
```
else if (condition)
{
    [...]
}
```